### PR TITLE
Input button component

### PR DESCRIPTION
**Pull request by Austrie**
Input button component: Including the regex check and error message feedback.

### CODE REVIEW

**Comment by mmw** [comment]
It's a little confusing why a new PR was created rather than commits added onto the PR where we were making feedback. Is that the typical workflow from the PC classes?

File: src/components/Input.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    this.setState({
+      inputText: e.target.value
+    }, this.validate())
+  }
+
+  validate = () => {
+    if (this.props.validation) {
+      const regex = new RegExp(this.props.validation)
+      if (regex.test(this.state.inputText)){
+        this.state.validated = true
```
Comment (mmw) [inline src/components/Input.js]: never mutate `state` directly, always use `this.setState`

File: src/components/Input.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      } else {
+        this.state.validated = false
+      }
+    }
+  }
+
+
+  drawErrorMessage = () => {
+    if (this.props.errorMessage) {
```
Comment (mmw) [inline src/components/Input.js]: could shorten this fn to

```
drawErrorMessage = () => this.props.errorMessage || "Invalid Input"
```

nitpicky comment, so feel free to ignore

**Comment by mmw** [comment]
update from dion: we do not want the `friend_referrral_code` and `referrral_code` to be namespaced under a year

**Comment by tlambrou** [comment]
@mmw Didn't realize you could add changes to the pull request. #thingsIDidn'tLearnAtMakeSchool.  I'll take a look at how to do this.  With that said, our main focus is to get an MVP up ASAP.

**Comment by alengel** [comment]
Oh yes, to add on to that, let me give you a handy checklist on how to use PRs. The workflow using pull requests should be something like this:

- Create a branch locally
- Have a bunch of commits pushed to that branch
- Push your branch up 
- Open a PR on Github using this neat UI
- Request a review from as many people as you like
- Wait for feedback / approval
- Use the feedback to commit some more things on the exact same branch 
=> if you don't agree with something or choose to ignore it, comment on the reviewers comments and let them know why (don't just ignore it!)
- Push it back up to Github
- Tag people using comments that your changes have been implemented and pushed back and ask for further feedback
- Keep doing that until you get approval for merge
- Merge it!

### Et voilá! You got some clean code in the codebase :)

**Comment by tlambrou** [comment]
@alengel this is great!  I'm adding to the Trello under resources.  THANK YOU!

**Pull request merged**